lua/lean/infoview.lua

```diff
@@ -118,8 +128,24 @@ function Infoview:focus_on_current_buffer()
   end
 end
 
---- Update this infoview's contents given the current position.
-function Infoview:update()
+function Info:new()
+  local new_info = {id = next_info_id, bufnr = vim.api.nvim_create_buf(false, true)}
+  infoview._info_by_id[next_info_id] = new_info
```

Comment:
It seems we never remove anything from `_info_by_id`, so if that's the case, it's just a growing array-like table and you may as well just get rid of `next_info_id` and use `table.insert` + `#_info_by_id`, or possibly just `vim.tbl_add_reverse_lookup` and have all the indexes and infos in one table.